### ISSUE TITLE
No errors or warnings found for button descendants of ARIA images

### ISSUE THREAD

**Issue by HadleyLuker** [validator/validator#2084]
### Spec URL

https://w3c.github.io/html-aria/#index-aria-img

### Test case

```html
<div role="img">
  <button type="button">Lorem ipsum</button>
</div>

<div role="img">
  <button type="submit">Dolor sit</button>
</div>

<div role="img">
  <button type="reset">Amet consectetur</button>
</div>

<div role="img">
  <input type="button" value="Adipiscing elit">
</div>

<div role="img">
  <input type="image" src="#" alt="Sed do">
</div>

<div role="img">
  <input type="reset" value="Eiusmod tempor">
</div>

<div role="img">
  <input type="submit" value="Incididunt ut">
</div>

<div role="img">
  <div role="button">Labore et</div>
</div>
```
